### PR TITLE
[backend] Create endpoints for managing tasks

### PR DESCRIPTION
**Pull request by vchrombie**
- [x] get_all_tasks
- [x] get_task
- [x] create_task
- [x] update_task
- [x] delete_task

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -1,15 +1,98 @@
...
+    return TrelloClient(api_key, token)
+
+client = get_trello_client(API_KEY, TOKEN)
+
+
 @app.get("/")
-def read_root():
-    return {"Hello": "World"}
+async def root():
+    # Retrieve all boards
```
Comment (soniasusanto) [inline backend/main.py]: Is this board the same concept as the list of todos? so the tasks below on line 52 will act as our todo item?

File: backend/main.py [hunk 2]
```diff
@@ -1,15 +1,98 @@
...
+    task = client.card.get_card(task_id)
+    return {
+        "message": f"Task {task_id} retrieved successfully.",
+        "task": task
+    }
+
+
+@app.post("/tasks/{task_id}")
+async def create_task(task_id: int, task: str = Body(...)):
+    # Create a new task
```
Comment (soniasusanto) [inline backend/main.py:70]: nice comments! makes the code easy to understand

File: backend/main.py (hunk 1 above)
Comment (vchrombie) [reply backend/main.py]: There is one board called `personal`, one list on the board called `doing` and there will be multiple tasks on that list.

**Review by soniasusanto — APPROVED** [review]
LGTM!

**Pull request merged**